rs/registry/routing_table/src/lib.rs

```diff
@@ -12,21 +12,9 @@ use std::{collections::BTreeMap, convert::TryFrom};
 pub const CANISTER_IDS_PER_SUBNET: u64 = 1 << 20;
 
 pub fn canister_id_into_u64(canister_id: CanisterId) -> u64 {
-    const LENGTH: usize = std::mem::size_of::<u64>();
-    let principal_id = canister_id.get();
-    let bytes = principal_id.as_slice();
-    // the +2 accounts for the two sentinel bytes that are added to the end of
-    // the array
-    assert_eq!(
-        bytes.len(),
-        LENGTH + 2,
-        "canister_id: {}; raw {:?}",
-        canister_id,
-        canister_id
-    );
-    let mut array = [0; LENGTH];
-    array[..LENGTH].copy_from_slice(&bytes[..LENGTH]);
-    u64::from_be_bytes(array)
+    canister_id
+        .to_u64()
```

Comment:
@alin-at-dfinity Is there any danger of the additional validation ( checking that the two bytes at the end are as expected that try_from_principal_id does) would cause any issues here?

Reply:
It may be an issue if someone tries to call this with a subnet principal. I don't see why they would knowingly do that, but we do shoehorn `SubnetIds` into `CanisterIds` when we want to route a message to a management canister (the type of the `Request` field is `CanisterId`, and we just take the `PrincipalId` within the `SubnetId` and wrap it in a `CanisterId` instead).

Reply:
I think we should figure out how to stop wrapping non-canisterId's in CanisterId (see other comment about Routable).